spec/lib/makefile_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'makefile'
+
+require 'tempfile'
+require 'open3'
+
+RSpec::Matchers.define :be_a_valid_makefile do
```

Comment:
This requires that `make` is installed in order to run tests. I can omit this if desired.